### PR TITLE
Support converting parameterized Qiskit circuit

### PR DESCRIPTION
**Pull request by liwt31**
Support converting parameterized Qiskit circuit. Can handle simple ParameterExpression such as `1.0 * theta` and will raise a ValueError if the expression is too complicated.

### CODE REVIEW

File: tensorcircuit/abstractcircuit.py [hunk 1]
```diff
@@ -711,6 +712,12 @@ def from_qiskit(
         :type n: int
         :param inputs: possible input wavefunction for ``tc.Circuit``, defaults to None
         :type inputs: Optional[List[float]], optional
+        :param circuit_params: circuit attributes such as the number of qubits
```
Comment (refraction-ray) [inline tensorcircuit/abstractcircuit.py:715]: , keys of dict as given in ``Circuit.__init__`` construction function

File: tensorcircuit/translation.py [hunk 2]
```diff
@@ -219,6 +219,47 @@ def qir2qiskit(
...
+                )
+            p_real = list(p.parameters)[0]
+            if not isinstance(p_real, ParameterVectorElement):
+                raise TypeError(
+                    "Parameters in parameter expression should be ParameterVectorElement"
+                )
+            # note "sym" != "sim"
+            expr = p.sympify().simplify()
+            # only allow simple expressions like 1.0 * theta
+            if not expr.is_Mul:
```
Comment (refraction-ray) [inline tensorcircuit/translation.py:245]: is there any common case for builtin qiskit ansatz beyond the simple 1*param style that you ever came across?

Comment (liwt31) [reply tensorcircuit/translation.py:245]: In UCCSD, the same parameter theta is used in multiple gates and there are both -1 * theta and 1 * theta. Converting all of them to 1 * theta will lose the phase information.

File: tensorcircuit/translation.py [hunk 3]
```diff
@@ -219,6 +219,47 @@ def qir2qiskit(
...
+                raise ValueError(f"Unsupported parameter expression: {p}")
+            arg1, arg2 = expr.args
+            if arg1.is_number and arg2.is_symbol:
+                coeff = arg1
+            elif arg1.is_symbol and arg2.is_number:
+                coeff = arg2
+            else:
+                raise ValueError(f"Unsupported parameter expression: {p}")
+            # taking real part here because using complex type will result in a type error
```
[the remark this answers is not in the document]
Comment (liwt31) [reply tensorcircuit/translation.py:254]: For me, I'd prefer converting to float due to its simple logic. Complex circuit parameter and gate parameter are rare. In fact, Qiskit does not support complex parameters at all.

File: tests/test_circuit.py [hunk 4]
```diff
@@ -1029,6 +1029,50 @@ def test_qiskit2tc():
...
+        from qiskit.circuit.library import TwoLocal
+        from qiskit_nature.second_q.circuit.library import UCCSD
+        from qiskit_nature.second_q.mappers import ParityMapper, QubitConverter
+    except ImportError:
+        pytest.skip("qiskit or qiskit-nature is not installed")
+    from tensorcircuit.translation import perm_matrix
+
+    mapper = ParityMapper()
+    converter = QubitConverter(mapper=mapper, two_qubit_reduction=True)
+    ansatz1 = UCCSD(2, [1, 1], converter)
```
[the remark this answers is not in the document]
Comment (liwt31) [reply tests/test_circuit.py:1046]: Both UCCSD and TwoLocal contain composed instructions and L413 is tested with parameters

File: tensorcircuit/abstractcircuit.py (hunk 1 above)
Comment (liwt31) [reply tensorcircuit/abstractcircuit.py:715]: OK

**Comment by liwt31** [comment]
> mypy failed in CI due to the generic type, eg. should at least be Dict[Any, Any] instead of Dict.

Thanks for the suggestion. Will fix.

> Other part looks good to me. Ultimately could add a jupyter tutorial on this nice feature with some quantum chemistry writeup (not necessary in this PR, and not necessary in this package if you have the tutorial plan in Chem package).

I'll probably do it in the chemistry package. It is indeed a nice feature worth documenting and not just for chemistry. I can add a few lines on the quick start page. But for a full tutorial, I'll prioritize writing it in the chemistry package first.

File: tensorcircuit/translation.py (hunk 2 above)
Comment (refraction-ray) [reply tensorcircuit/translation.py:245]: good point to against parameterexpression -> jax tensor idea! the connection between parameters will lose

File: docs/source/quickstart.rst [hunk 5]
```diff
@@ -81,7 +81,9 @@ We currently support transform ``tc.Circuit`` from and to Qiskit ``QuantumCircui
 
 Export to Qiskit (possible for further hardware experiment, compiling, and visualization): ``c.to_qiskit()``.
 
-Import from Qiskit: ``c = tc.Circuit.from_qiskit(QuantumCircuit, n)``
+Import from Qiskit: ``c = tc.Circuit.from_qiskit(QuantumCircuit, n)``.
+Parameterized Qiskit circuit is supported by passing the parameters to the ``from_qiskit`` function,
+similar to the ``assign_parameters`` function in Qiskit.
```
Comment (refraction-ray) [inline docs/source/quickstart.rst:86]: at least mention the corresponding argument name in tc ``bind_params``?

**Comment by refraction-ray** [comment]
also pylint error reported by CI as below

************* Module tensorcircuit.abstractcircuit
tensorcircuit/abstractcircuit.py:[7](https://github.com/tencent-quantum-lab/tensorcircuit/actions/runs/3879849333/jobs/6617408196#step:7:8)29:0: C0301: Line too long (139/120) (line-too-long)
************* Module tests.test_circuit
tests/test_circuit.py:1036:[8](https://github.com/tencent-quantum-lab/tensorcircuit/actions/runs/3879849333/jobs/6617408196#step:7:9): W0611: Unused Statevector imported from qiskit.quantum_info (unused-import)

**Review by refraction-ray — APPROVED** [review]
LGTM

**Pull request merged**